suite-common/wallet-utils/src/__tests__/accountUtils.test.ts

```diff
@@ -337,4 +340,119 @@ describe('account utils', () => {
+        expect(isDebugOnlyAccountType('segwit', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('coinjoin', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('taproot', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('ledger', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('legacy', 'eth')).toBe(true);
+        expect(isDebugOnlyAccountType('ledger', 'eth')).toBe(true);
+    });
+
+    type RunFilterReceiveAccountsTestParams = {
+        isDebug?: boolean;
+        expectedAccounts?: Account[];
+        receiveNetwork?: string;
+        deviceState?: `${string}@${string}:${number}`;
+    };
+
+    const runFilterReceiveAccountsTest = ({
+        isDebug = true,
+        expectedAccounts = [],
+        receiveNetwork = 'eth',
+        deviceState = '1stTestnetAddress@device_id:0',
+    }: RunFilterReceiveAccountsTestParams) => {
+        const device = getSuiteDevice({
+            unavailableCapabilities: {
+                dash: 'no-support',
+            },
+            state: deviceState,
+        });
+        const unavailableCapabilities = device?.unavailableCapabilities ?? {};
+
+        const receiveNetworks = networksCollection.filter(
+            (n: Network) =>
+                n.symbol === receiveNetwork &&
+                !unavailableCapabilities[n.symbol] &&
+                ((n.isDebugOnlyNetwork && isDebug) || !n.isDebugOnlyNetwork),
+        );
+
+        const accounts = [
+            getWalletAccount({ symbol: 'eth', accountType: 'legacy' }),
+            getWalletAccount({ symbol: 'eth', accountType: 'normal' }),
+            getWalletAccount({ symbol: 'eth', accountType: 'ledger' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'coinjoin' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'taproot' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'legacy' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'segwit' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'ledger' }),
+            getWalletAccount({ symbol: 'matic', accountType: 'legacy' }),
+            getWalletAccount({ symbol: 'matic', accountType: 'normal' }),
+            getWalletAccount({ symbol: 'matic', accountType: 'ledger' }),
+            getWalletAccount({ symbol: 'sol', accountType: 'normal', empty: true, visible: false }),
+            getWalletAccount({ symbol: 'sol', accountType: 'ledger' }),
+            getWalletAccount({
+                symbol: 'sol',
+                accountType: 'ledger',
+                empty: true,
+                visible: false,
+            }),
+        ];
+
+        expect(
+            filterReceiveAccounts({
+                accounts,
+                deviceState: device.state,
+                receiveNetwork,
+                isDebug,
+                receiveNetworks,
+            }),
+        ).toEqual(expectedAccounts);
+    };
+
+    it('filterReceiveAccounts: isDebug', () => {
```

Comment:
to be fair, other its in this file are labeled similarly, for example `enhanceAddresses`.
I don't mean to to defend it, I think this file could do with a cleanup, I'd like to see one `describe` per one tested function instead of  `describe('account utils'`.

But this PR is consistent with current state :shrug:

Reply:
I think I will leave the descriptions how they are for the consistency with the file, but it's a good point and we can do some refactoring later